### DISCUSSION TITLE
Add plug-in panel function

### DISCUSSION THREAD

**Discussion by ghost** [nvaccess/nvda#14370] [General]
# question

　　As the number of software plug-ins increases, the number of shortcuts that need to be assigned is also increasing.The numerous shortcut keys are hard to remember.

# New Features

　　We can imagine a new feature.By adding a new panel, we can add the plug-in functions that we want to assign shortcuts to.This allows us to add functionality to the panel when we come across something that might be useful but isn't worth assigning a shortcut to.Although there are some functions based on the current text and the current object that are not available in this case, this is not a problem.

# Explain in detail

　　The functions we added in this panel are some switches and functions that do not need to be implemented based on the current object and text.For example, a plug-in provides the function of switching mute and unmute. Then we can add this function to the panel, which will save a shortcut key.Another example is the copy link function of the remote assistance plug-in.

# Extend

　　We can even extend the panel to some extent, which can also improve efficiency.For example, we can classify the added functions. In this way, you can find the function you want very quickly and accurately.

I'm very sorry that I can't express my thoughts exactly in English. Because I come from China, I use translation software to write.

**Comment by josephsl** [comment]
Hi,

Are you talking about a panel where you can assign keyboard shortcuts to NVDA add-on features, or something else? If you are talking about keyboard shortcuts, it is already available: input gestures dialog in NVDA menu/Preferences.

Thanks.

**Comment by rperez030** [comment]
I believe the feature request refers to a panel which can be invoked everywhere, kind of like the Python console, from which we can browse an execute gestures that may not have a shortcut associated to them or in situations in which we don't remember the shortcut. I believe something like that could help reduce cognitive load and make NVDA functionality more discoverable. It could also be a great way to access scripts exposed by add-ons that do not have a default shortcut.

**Comment by ghost** [comment]
> Hi,
> 
> Are you talking about a panel where you can assign keyboard shortcuts to NVDA add-on features, or something else? If you are talking about keyboard shortcuts, it is already available: input gestures dialog in NVDA menu/Preferences.
> 
> Thanks.

I'm not talking about assigning shortcuts to plugin functions.The former, I mean, is to create a panel and add the desired plug-in functionality to it.Because if you assign shortcuts to every plug-in, it's hard to remember.There are some plug-in functions that are occasionally used, but it is very difficult to remember his shortcut keys. If there is such a panel, you can find it directly, and you don't need to remember these shortcut keys any more.

**Comment by ghost** [comment]
The embarrassing thing is that I can't express my thoughts accurately in English.I also posted about it on a Facebook group, and then the developer asked me to submit an issue.

Please forgive my poor translation.

I don't mean to add shortcuts to plugin functions or change the shortcuts of plugin functions, which are already there.

To be precise, what I mean is to add a panel to add some plug-in functions that we don't often use.In this way, we can add it when we forget its shortcut key or don't want this function to occupy our shortcut key.

Thank you very much for reading.

**Comment by mzanm** [comment]
Do you mean like a dialog that allows triggering any script of an add-on regardless if it's assigned to a gesture or not? Similar to narrators command list narrator + f 1 except you could customize the list with what to show rather than it showing all commands?

**Comment by ghost** [comment]
> Do you mean like a dialog that allows triggering any script of an add-on regardless if it's assigned to a gesture or not? Similar to narrators command list narrator + f 1 except you could customize the list with what to show rather than it showing all commands?

Yes, that's what it means.

We can add any function that can be used with shortcut keys to this panel.In this way, when we don't have enough shortcut keys, or when we forget the shortcut keys, we can use this function through this panel.We can even set categories for the panel to make it easier to find.

**Comment by paulber19** [comment]
This feature was implemented in the NVDAExtensionGlobalPlugin add-on and 
consisted of adding a "Run script" button in the "Input gestures" dialog.
This button causes the execution of the selected script, even if no 
input gesture is assigned to it.
This is indeed a feature that should be at the NVDA core.
